client/signup/steps/page-picker/shopping-cart-for-difm.tsx

```diff
@@ -0,0 +1,206 @@
+			display: flex;
+			font-size: 0.9em;
+		}
+	}
+	.page-picker__price {
+		font-weight: 500;
+	}
+	.page-picker__meta {
+		color: #646970;
+		font-size: 12px;
+		width: 100%;
+		display: flex;
+		flex-direction: row;
+		align-content: center;
+		justify-content: space-between;
+		flex-wrap: wrap;
+		gap: 2px 10px;
+	}
+	span {
+		position: relative;
+	}
+
+	@media ( max-width: 600px ) {
+		font-size: 0.9em;
+		padding: 2px 0;
+		border: none;
+
+		.page-picker__meta {
+			display: none;
+		}
+	}
+`;
+
+const LineItemsWrapper = styled.div`
+	box-sizing: border-box;
+	margin: 20px 0;
+	padding: 0;
+	overflow-y: auto;
+	max-height: 75vh;
+	width: 100%;
+	max-width: 365px;
+	border-bottom: 1px solid #dcdcde;
+	@media ( max-width: 600px ) {
+		margin: 0 0px;
+		border: none;
+		max-width: 100%;
+	}
+`;
+
+const Total = styled.div`
+	font-weight: 500;
+	display: flex;
+	flex-wrap: wrap;
+	justify-content: space-between;
+	padding: 30px 0;
+	div {
+		display: flex;
+		align-items: center;
+		&.value {
+			font-family: Recoleta;
+			font-size: 32px;
+			font-weight: 400;
+			@media ( max-width: 600px ) {
+				font-size: 16px;
+			}
+		}
+	}
+
+	@media ( max-width: 600px ) {
+		font-size: 16px;
+		margin-top: 6px;
+		padding: 0;
+		padding-top: 6px;
+		border-top: 1px solid #eee;
+	}
+`;
+
+function DummyLineItem( {
+	product,
+	meta,
+	productCount,
+	nameOverride,
+	currencyCode,
+}: CartItem & { currencyCode: string } ) {
+	if ( ! product ) {
+		return null;
+	}
+
+	return (
+		<DummyLineItemContainer>
+			<div className="page-picker__title">{ nameOverride ?? product.product_name }</div>
+			<div className="page-picker__price">
+				{ productCount !== undefined
+					? formatCurrency( product.cost * productCount, currencyCode, { precision: 0 } )
+					: formatCurrency( product.cost, currencyCode, { precision: 0 } ) }
+			</div>
+			{ meta && <div className="page-picker__meta">{ meta }</div> }
+		</DummyLineItemContainer>
+	);
+}
+
+export default function ShoppingCartForDIFM( { selectedPages }: { selectedPages: string[] } ) {
+	const translate = useTranslate();
+	const { items, total, isCartLoading } = useCartForDIFM( selectedPages );
+	const signupDependencies = useSelector( getSignupDependencyStore );
+	const { newOrExistingSiteChoice } = signupDependencies;
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+
+	return newOrExistingSiteChoice === 'existing-site' && isCartLoading ? (
+		<LoadingContainer>
+			{ items.length === 0 ? (
+				<>
+					<LoadingLine key="plan-placehorder" />
+					<LoadingLine key="difm-placehorder" />
+				</>
+			) : (
+				items.map( ( lineItem ) => <LoadingLine key={ lineItem.product.product_slug } /> )
+			) }
+			<LoadingLine key="total-placehorder" />
+		</LoadingContainer>
+	) : (
+		<CartContainer>
+			<Cart>
+				<LineItemsWrapper>
+					{ items.map( ( item ) => (
+						<DummyLineItem
+							key={ item.product.product_slug }
+							{ ...item }
+							currencyCode={ currencyCode as string }
```

Comment:
If for some reason a currency is not set in Redux's `state.state.currencyCode`, then `currencyCode` here will be `undefined`. I looked to see what effect that might have and it looks like it would be passed to `formatCurrency()`, which says it requires a string. I looked deeper, and [that is used to determine the currency data](https://github.com/Automattic/wp-calypso/blob/b04ef247c852fd5068227f55ed3fa90ce6c12295/packages/format-currency/src/currencies.ts#L978-L987), defaulting to use the settings for USD. 

So this type cast (`as string`) is _implicitly_ saying, "use USD if we do not have a currency set". 

I think it would be better to remove the type cast and _explicitly_ set the default to USD. 
```suggestion
							currencyCode={ currencyCode ?? 'USD' }
```

That way if someone saw this code in the future it would be clear what is happening without needing to dig into the source of the formatting package.